### PR TITLE
modernize `conda_install` test on Circle CI by installing from conda-forge with Python 3.9 and change install instructions in documentation

### LINKED ISSUE #1279: `conda_install` test on Circle CI is failing often running out of available machine memory
[bummer example](https://app.circleci.com/pipelines/github/ESMValGroup/ESMValCore/5450/workflows/70af57ee-da6a-4ddc-9cb9-b85c51439234/jobs/26109) - am looking into it now

### PR DESCRIPTION
**Pull request by valeriupredoi**
## Description

Closes #1279 
[conda install](https://app.circleci.com/pipelines/github/ESMValGroup/ESMValCore/5473/workflows/ee6c1e56-a960-4200-a2ab-b28502b8a8b0/jobs/26229) works fine with these changes, hopefully it'll not run out of memory anymore, at least as often as before :beer: 

***

## [Before you get started](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#getting-started)

- [ ] [☝ Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do

## [Checklist](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#checklist-for-pull-requests)

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🧪][2] The new functionality is [relevant and scientifically sound](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#scientific-relevance)
- [x] [🛠][1] This pull request has a [descriptive title and labels](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-title-and-label)
- [x] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-checks) were successful

***

To help with the number pull requests:

- 🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValCore/pulls) in this repository

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -208,10 +208,10 @@ jobs:
             mkdir /logs
             # conda update -y conda > /logs/conda.txt 2>&1
             # Create and activate conda environment
-            conda create -y --name esmvaltool
+            conda create -y --name esmvaltool 'python=3.9'
```
Comment (bouweandela) [inline .circleci/config.yml:211]:
```suggestion
            conda create -y --name esmvaltool -c conda-forge 'python=3.9'
```
or it's going to create an environment full of packages from the defaults channel that then need to be uninstalled in the next step

**Review by bouweandela — CHANGES_REQUESTED** [review]
Thanks for taking care of this! Could you also check that these steps are identical to the documentation?

File: .circleci/config.yml (hunk 1 above)
Comment (valeriupredoi) [reply .circleci/config.yml]: very good point! Overlooked it :eyes:

**Comment by valeriupredoi** [comment]
> Thanks for taking care of this! Could you also check that these steps are identical to the documentation?

no, good point, man - we need to remove the use of `esmvalgroup` channel for Core in the documentation, should I do that here or in a different PR (I'd do it in a separate PR right away if it was by me)

**Comment by bouweandela** [comment]
It would be great if you could do it here, the tests on CircleCI/GitHub Actions should always reflect the documentation.

**Comment by valeriupredoi** [comment]
right on @bouweandela - will do it 'ere then :+1:

**Comment by valeriupredoi** [comment]
OK Bouwe, done that, could you have a looksee pls mate, I've also changed a bit the instructions on how to build&upload the package :beer:

File: doc/quickstart/install.rst [hunk 2]
```diff
@@ -14,15 +14,15 @@ Once you have installed conda, you can install ESMValCore by running:
...
+    conda install -c conda-forge esmvalcore
 
 It is also possible to create a new
 `Conda environment <https://docs.conda.io/projects/conda/en/latest/user-guide/getting-started.html#managing-environments>`_
 and install ESMValCore into it with a single command:
 
 .. code-block:: bash
 
-    conda create --name esmvalcore -c esmvalgroup -c conda-forge esmvalcore
+    conda create -y --name esmvaltool -c conda-forge 'python=3.9'
```
Comment (bouweandela) [inline doc/quickstart/install.rst:25]:
```suggestion
    conda create --name esmvalcore -c conda-forge esmvalcore 'python=3.9'
```
or the new environment will only contain python and not esmvalcore. Do you think we should change the name of the environment too?

Comment (bouweandela) [reply doc/quickstart/install.rst:25]: Also, I think it is important that the instructions are the same as what we do on CircleCI, but this is different

Comment (valeriupredoi) [reply doc/quickstart/install.rst]: man, I keep eating important syntax today :rofl: Hmmm I think it's fine `esmvaltool` - one can start getting confused if we change names, I already had a case of a student using two different environments each for Core and Tool

Comment (valeriupredoi) [reply doc/quickstart/install.rst]: > Also, I think it is important that the instructions are the same as what we do on CircleCI, but this is different

This is what we do on Circle CI, the problem is that we don't test other bits of the instructions, like the direct `conda install -c conda-forge esmvalcore` call (but I think we test that in GA)

Comment (valeriupredoi) [reply doc/quickstart/install.rst]: and of course I changed the env name :man_facepalming: - I need a beer :grin:

Comment (valeriupredoi) [reply doc/quickstart/install.rst]: oh no I didn't, but I still need a beer :beer:

**Review by bouweandela — APPROVED** [review]

**Comment by valeriupredoi** [comment]
cheers @bouweandela :beer:

**Pull request merged**